indico_toolkit/association/line_items.py

```diff
@@ -35,31 +36,28 @@ def __init__(
         line_item_fields (Iterable[str]): Fields/labels to include as line item values, other values
                                       will not be assigned a row_number. If None, treats all fields 
                                       as line item fields.
-
         """
-        if isinstance(predictions, Predictions):
-            predictions = predictions.to_list()
+        self.predictions = self.validate_prediction_formatting(predictions)
         if line_item_fields is None:
-            print("No line item fields provided. Will treat all predictions as line items")
-            line_item_fields = Predictions.get_extraction_labels_set(predictions)
+            print(
+                "No line item fields provided. Will treat all predictions as line items"
```

Comment:
this is assuming that the user's intention - in not passing line_item_fields - is that they have pre-filtered the predictions down to those that need to be grouped?  i think i would prefer to force the user to feed all predictions + line_item_fields list.

Reply:
making this change